task3/DynamicArray/DynamicArray.cs

```diff
@@ -0,0 +1,208 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DynamicArray
+{
+    public class DynamicArray<T> : IEnumerable<T>, IEnumerable, ICloneable
+    {
+        private T[] _array;
+        public int Length { get; private set; }
+        public int Capacity
+        {
+            get => _array.Length;
+            set
+            {
+                ChangeCapacity(value);
+            }
+        }
+        /// <summary>
+        /// this method allows you to change the capacity of the array. If you set the capacity less 
+        /// than the current one, some of the data will be deleted
+        /// </summary>
+        /// <param name="value">new capacity</param>
+        private void ChangeCapacity(int value)
```

Comment:
I get that it's a method for a property, but still, member types order matters - properties with properties, methods with methods.